boards/PDM/Inc/Io/Io_Efuse.h

```diff
@@ -0,0 +1,726 @@
+    } faults;
+    struct EfuseChipSelect
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } chip_select;
+    struct EfuseCurrentSenseSync
+    {
+        GPIO_TypeDef *GPIO_Port;
+        uint16_t      GPIO_Pin;
+    } current_sense_sync;
+};
+
+/**
+ * Initializes the SPI handle for the Efuses
+ * @param hspi Handle to the SPI peripheral used for the Efuses
+ */
+void Io_Efuse_Init(SPI_HandleTypeDef *const hspi);
+
+/**
+ * Configures the Efuse's registers using the values defined in:
+ * REGISTERNAME_CONFIG
+ * @param e_fuse Pointer to Efuse structure being configured
+ */
+void Io_Efuse_ConfigureEfuse(struct Efuse *e_fuse);
+
+/**
+ * Sets current/temperature monitoring option of CSNS pin
+ * @param selection The desired monitoring function which can be selected by
+ * specifying the following values: CSNS_FUNCTION_DISABLED - Current sensing
+ * disabled CSNS_FUNCTION_CURRENT_CH0 - Current sensing for channel 0
+ *                  CSNS_FUNCTION_CURRENT_CH1 - Current sensing for channel 1
+ *                  CSNS_FUNCTION_TEMPERATURE - Ffuse temperature sensing
+ *                  CSNS_FUNCTION_CURRENT_SUM - Current sensing for summed
+ * channels (for Parallel mode)
+ * @param e_fuse Pointer to Efuse structure being configured
+ */
+void Io_Efuse_Configure_Channel_Monitoring(
+    uint8_t       selection,
+    struct Efuse *e_fuse);
+
+/**
+ * Gets the value of the Efuse's Status Register over SPI and updates the Efuse
+ * handle's operating status
+ * @param e_fuse Pointer to Efuse structure being checked
+ */
+void Io_Efuse_UpdateStatus(struct Efuse *e_fuse);
+
+/**
+ * Get the value of the Efuse's Fault Register over SPI and updates the Efuse
+ * handle's fault status handle's fault status
+ * @param e_fuse Pointer to Efuse structure being checked
+ */
+void Io_Efuse_UpdateFaults(struct Efuse *e_fuse);
+
+/**
+ * Write data to a specific Serial Input register on the Efuse
+ * @param register_address Serial Input register being written to
+ * @param register_value Value being written to the Serial Input register
+ * @param e_fuse Pointer to Efuse structure being written to
+ */
+void Io_Efuse_WriteReg(
+    uint8_t       register_address,
+    uint16_t      register_value,
+    struct Efuse *e_fuse);
+
+/**
+ * Read data from a specific Serial Output register on the Efuse
+ * @param register_address Serial Output register being read from
+ * @param e_fuse Pointer to Efuse structure being read from
+ * @return Contents of the addressed register
+ */
+uint16_t Io_Efuse_ReadReg(uint8_t register_address, struct Efuse *e_fuse);
+
+/**
+ * Calculates the parity of the SPI command to be sent to the Efuse.
+ * Using the XOR sum of bits method taken from:
+ * https://en.wikipedia.org/wiki/Parity_bit#Parity
+ * @param spi_command Original SPI command without the parity bit
+ * @return Returns true if the parity bit needs to be set
+ */
+bool Io_Efuse_Set_Parity_Bit(uint16_t spi_command);
```

Comment:
```suggestion
bool Io_Efuse_SetParityBit(uint16_t* spi_command);
```

Honestly I think you should just pass-by-reference because this function is called "Set" parity bit